YAXLib/XMLUtils.cs

```diff
@@ -72,6 +73,21 @@ public static XElement FindLocation(XElement baseElement, string location)
             return currentLocation;
         }
 
+#nullable enable
+        /// <summary>
+        /// Strips all invalid characters from the input value, if <paramref name="enabled"/> is <see langword="true"/>.
+        /// </summary>
+        /// <param name="input"></param>
+        /// <param name="enabled"></param>
+        /// <returns></returns>
+        public static string StripInvalidXmlChars(this string? input, bool enabled)
+        {
+            return enabled && input != null
```

Comment:
For performance reasons you could first check if the string contains an invalid char 

(Creating a string from a char array is a bit expensive)

Reply:
Will use `ArrayPool<char>` to remove invalid chars